subsys/net/ip/Kconfig

```diff
@@ -431,6 +431,16 @@ config NET_TCP_MAX_SEND_WINDOW_SIZE
 	  size. The default value 0 lets the TCP stack select the value
 	  according to amount of network buffers configured in the system.
 
+config NET_TCP_MAX_RECV_WINDOW_SIZE
+	int "Maximum receive window size to use"
+	depends on NET_TCP
+	default 1280
```

Comment:
I just wish that the default value could be calculated at runtime from amount of available network buffers but this is a good start.
Shouldn't we also implement `SO_RCVBUF` so that it is possible to change the max value at runtime?

Reply:
I could implement similar logic to send window - would the same formula to calculate the window size `(CONFIG_NET_BUF_RX_COUNT * CONFIG_NET_BUF_DATA_SIZE) / 3;` be acceptable for receive window as well?

As for `SO_RCVBUF`, I'd leave it out of this PR as for now not to overcomplicate things, we can add it later if needed.

Reply:
> I could implement similar logic to send window - would the same formula to calculate the window size `(CONFIG_NET_BUF_RX_COUNT * CONFIG_NET_BUF_DATA_SIZE) / 3;` be acceptable for receive window as well?

Sounds good, then the default value is somewhat related to allocated resources and we can override it from config if needed.

> As for `SO_RCVBUF`, I'd leave it out of this PR as for now not to overcomplicate things, we can add it later if needed.

Yep, I was thinking the same.